### ISSUE TITLE
VerifyingOps with named argument method invocation cause unused-locals warning

### ISSUE THREAD

**Issue by chikei** [mockito/mockito-scala#371]
```scala
import java.time.Instant
trait TestTrait {
  def anyRef(a: Instant, b: Instant): Unit
}

val t = mock[TestTrait]
t.anyRef(b = Instant.EPOCH, a = Instant.EPOCH).wasCalled(once)
```
results some cryptic messages like `local val x$3 in value <local $anon> is never used`.

Poking around compiler settings, it seems due to the fact normally scala will desugar named argument into
```scala
  val x$1: java.time.Instant = java.time.Instant.EPOCH;
  val x$2: java.time.Instant = java.time.Instant.EPOCH;
  t.anyRef(x$2, x$1)
```
but VerifyingOps and related macro will expand into
```scala
  <artifact> val x$3: java.time.Instant = java.time.Instant.EPOCH;
  <artifact> val x$4: java.time.Instant = java.time.Instant.EPOCH;
  verification(mockito.this.VerifyOrder.unOrdered.verifyWithMode($anon.this.t, TestSpec.this.once)
    .anyRef(_root_.org.mockito.matchers.DefaultMatcher(java.time.Instant.EPOCH),
      _root_.org.mockito.matchers.DefaultMatcher(java.time.Instant.EPOCH)
    )
  )
```

note that linter in scalac seems have some special rule for primitives so Int/Boolean/String etc won't cause the warning.

**Comment by ultrasecreth** [comment]
Interesting, I thought we had tests that verified this compiler extraction of arguments work, I'll have a look

**Comment by ultrasecreth** [comment]
First of all, I can't reproduce the issue, not even adding `-Ywarn-unused:locals` to my compiler settings, so I'm flying a bit blind here.

I can see the unused `val`s left behind by the macro, so I've made some changes based on what I can see, but I can't be sure I've done all the necessary changes (although I think I did) as I could never get the compiler to fail for me, so please get the version published after PR #372 gets merged and give me some feedback please.

**Comment by chikei** [comment]
I just realized it's caused by scalac option `-Wmacros:after`, sorry for forgot mention that. I've run `sbt scalatest/test:compile` with the compiler option enabled against PR #372 and it compiles.
